feat: send emails about successfull verification

nadiia-tsytsylina opened the pull request:
## Describe your changes
 add function for sending emails to successfully verified users, depending on role
 
## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/AC-81)

Caregiver email:
<img width="800" alt="image" src="https://github.com/ZenBit-Tech/ctrlchamps_be/assets/117380986/6c04be1e-ba90-4282-a134-b7de9b34718d">

Seeker email:
<img width="793" alt="image" src="https://github.com/ZenBit-Tech/ctrlchamps_be/assets/117380986/d71b7f14-2f28-4cc9-8672-9ce0d251861d">


### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [ ] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

nadiia-tsytsylina committed:
feat: send emails about successfull verification

nadiia-tsytsylina committed:
fix: clear code

Simpsonkk commented on src/modules/auth/auth.service.ts:
```diff
@@ -190,6 +208,40 @@ export class AuthService {
     }
   }
 
+  private async sendVerifiedEmail(email: string, role: string): Promise<void> {
```
Please, get rid of duplication in this method.

nadiia-tsytsylina replied:
thank you. fixed

nadiia-tsytsylina committed:
fix: deleted code duplication

Simpsonkk commented:
Approved.

nadiia-tsytsylina committed:
Merge branch 'develop' into features/account-verification-emails

nadiia-tsytsylina committed:
fix: add values to .env

nadiia-tsytsylina committed:
Merge branch 'features/account-verification-emails' of https://github.com/ZenBit-Tech/ctrlchamps_be into features/account-verification-emails

lyudad commented:
<img width="510" alt="Screenshot 2023-11-23 at 22 19 49" src="https://github.com/ZenBit-Tech/ctrlchamps_be/assets/25103090/48b98895-a22f-468c-a272-a63488c1f7e3">

lyudad commented:
<img width="308" alt="Screenshot 2023-11-23 at 22 20 26" src="https://github.com/ZenBit-Tech/ctrlchamps_be/assets/25103090/bee9ab7e-42a5-483a-9b21-7a9d53e7a81c">

nadiia-tsytsylina commented:
> <img alt="Screenshot 2023-11-23 at 22 19 49" width="510" src="https://user-images.githubusercontent.com/25103090/285301526-48b98895-a22f-468c-a272-a63488c1f7e3.png">

Can you please clarify what exactly should be corrected in the text?
CrtrlChamps instead of Afyanex Care? Or delete "team" from the last row?

Simpsonkk commented:
> > <img alt="Screenshot 2023-11-23 at 22 19 49" width="510" src="https://user-images.githubusercontent.com/25103090/285301526-48b98895-a22f-468c-a272-a63488c1f7e3.png">
> 
> Can you please clarify what exactly should be corrected in the text? CrtrlChamps instead of Afyanex Care? Or delete "team" from the last row?

Should be our team name

fedasevich commented on src/common/enums/error-message.enum.ts:
```diff
@@ -9,4 +9,5 @@ export enum ErrorMessage {
   UserNotExist = "An account matching that email doesn't exist",
   UserEmailAndPhoneAlreadyExists = 'User with these email and phone already exists',
   BadLoginCredentials = 'Failed to sign in due to bad credentials',
+  FailedSendSuccessEmail = 'Failed to send email',
```
we already have that error in email module enum

nadiia-tsytsylina replied:
fixed

nadiia-tsytsylina commented:
fix: 
deleted unsubscribe, 
updated text,
deleted unnecessary error message,
updated .env

![image](https://github.com/ZenBit-Tech/ctrlchamps_be/assets/117380986/a9e92e73-3218-42fc-8b64-f6e847d373d9)

![image](https://github.com/ZenBit-Tech/ctrlchamps_be/assets/117380986/82b806f5-9e30-4631-a296-05976304ccd2)

nadiia-tsytsylina committed:
fix: deleted unnesessary error message, upd env

nadiia-tsytsylina committed:
Merge branch 'develop' into features/account-verification-emails

lyudad commented:
кстати текста так и не поменялись(

nadiia-tsytsylina commented:
> кстати текста так и не поменялись(

Я не поняла, что именно нужно поменять. Посоветовалась с Любой и убрала слово "team" в конце, где было Afyanex Care Team. Нужно было поменять Afyanex на CtrlChamps?

nadiia-tsytsylina committed:
Merge branch 'develop' into features/account-verification-emails
